### PR TITLE
Spacecraft file tutorial

### LINKED ISSUE #147: Reorganize Point_source_resonse.ipynb (to be called SpacecraftFile.ipynb) and DetectorResponse.ipynb tutorials.
Following up from PR #145 

- What is currently called "Point_source_resonse.ipynb" will change names to "SpacecraftFile.ipynb" and focus on the coordinates, orientation, etc. (described in 2. above)
- I'll add the explanation of the scatt map to SpacecraftFile.ipynb
- @Yong2Sheng will move the actual point source calculation to “DetectorResponse.ipynb” (that is, replacing my code with a fake orientation history)
- PR #145  must be merged first

### PR DESCRIPTION
**Pull request by israelmcmc**
Closed #147 

I'll track my changes here to coordinate with @Yong2Sheng

- Changed the name of the nb
- Removed mention of deprecated formats. We only use the .ori files right now.
- I merged all of @Yong2Sheng  commits from #150
- @Yong2Sheng : good ideas to get the unique pixels during the dwell time map calculation! That inspired me to update a few lines to make it even faster!
- dts in SpacecraftFile is time a TimeDelta object, not Time (as it should).
- I did a general overhaul of the SpacecraftFile and DetectorResponse tutorials. I added information, better plots, reorganized the text, etc.

### CODE REVIEW

**Comment by israelmcmc** [comment]
@Yong2Sheng Can you please check and merge this PR? In particular, that both tutorials work, that I didn't remove any information you wanted, and that SpacecraftFile.py still works well.

**Comment by Yong2Sheng** [comment]
Hi @israelmcmc,

I reviewed the PR, it looks great! I really like your algorithm to get the summed pixel durations.

I tested [DetectorResponse.ipynb](https://github.com/israelmcmc/cosipy/blob/scatt_map_docs/docs/tutorials/response/DetectorResponse.ipynb) and [SpacecraftFile.ipynb](https://github.com/israelmcmc/cosipy/blob/scatt_map_docs/docs/tutorials/response/SpacecraftFile.ipynb). They both ran very well.

I only have one minor comment on one line I posted above. 

Thank you very much!

File: cosipy/spacecraftfile/SpacecraftFile.py [hunk 1]
```diff
@@ -398,14 +402,37 @@ def get_dwell_map(self, response, dts = None, dt_format = None, src_path = None,
...
-                for p, w in zip(pixels, weights):
-                    self.dwell_map[p] += w*(duration.unix*u.s)
+        # Get the unique pixels to weight, and sum all the correspondint weights first, so
+        # each pixels needs to be called only once.
+        # Based on https://stackoverflow.com/questions/23268605/grouping-indices-of-unique-elements-in-numpy
+        
+        # remove the last value. Effectively a 0th order interpolations
+        pixels, weights = self.dwell_map.get_interp_weights(theta = self.src_path_skycoord[:-1])  
 
+        weighted_duration = weights * self.dts.to_value(u.second)[None]
```
[the remark this answers is not in the document]
Comment (israelmcmc) [reply cosipy/spacecraftfile/SpacecraftFile.py:414]: @Yong2Sheng At this point "weighted_duration" is still a 2D matrix. I'm using broadcasting, but you had the equivalent code too:
```
        dts_matrix = np.row_stack((self.dts.value,)*4)  # create the dts matrix to multiply the weights
        weighted_duration = weights * dts_matrix
```
This is just a tiny bit faster, but it's the same calculation.

**Comment by israelmcmc** [comment]
Note: all commits from PR #150 were merged here.

**Review by Yong2Sheng — APPROVED** [review]
It looks good!

**Review by Yong2Sheng — APPROVED** [review]

**Review by Yong2Sheng — APPROVED** [review]

**Review by Yong2Sheng — APPROVED** [review]

**Review by Yong2Sheng — APPROVED** [review]

**Pull request merged**